### DISCUSSION TITLE
How to configure .wezterm.lua to interrogate for existing gui on startup and spawn a new tab vs starting a new window

### DISCUSSION THREAD

**Discussion by tmknight** [wezterm/wezterm#4946] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### WezTerm version

20240128-202157-1e552d76

### Ask your question!

I've searched and reviewed documentation, however, it escapes me how this might be accomplished.  I know about `wezterm cli spawn`, but I want this to be dynamic, automatic and default behavior.  Seems it should be possible within the lua config.

if wezterm is already running and wezterm is invoked again, I simply want wezterm to spawn a new tab in an existing window .

E.g.  I have wezterm running with one ( or more, doesn't matter in this use-case) tab open.  I invoke wezterm again (say via open terminal here in windows manager), I want a new tab to be created in that existing window (preferably as the last tab (from left to right)).  If wezterm isn't running then, the open here should start wezterm for the first time.

I can image there would be a need to override this behavior when I do want an additional window, so that would need to be built into the configuration.

Any help the community can offer would be greatly appreciated.

Cheers

**Comment by wez** [comment, marked as answer]
Nightly builds now have `wezterm start --new-tab` and `wezterm connect --new-tab` to request a new tab rather than a new window when spawning via an existing GUI instance. The new [prefer_to_spawn_tabs](https://wezfurlong.org/wezterm/config/lua/config/prefer_to_spawn_tabs.html) option allows you to make this happen by default. [?4854](https://github.com/wez/wezterm/discussions/4854) [?4946](https://github.com/wez/wezterm/discussions/4946)

**Comment by tmknight** [comment]
That's great news.  Though invoking via cli and setting via config (I apply via `wezterm.prefer_to_spawn_tabs = true`, correct? ) do 
not yield the expected result (nightly installed via nightly RPM `20240203-080655-7d84bf5b`).

A new window opens when invoked via external call such as open terminal here or `wezterm start --new-tab`

**Reply by wez** [reply]
when you have `config.prefer_to_spawn_tabs = true` set in your config, running `wezterm`, `wezterm spawn` or launching it via its icon should cause it to spawn a new tab rather than a new window.

**Comment by tmknight** [comment]
Thanks, I had initially installed 20240202 from copr, but then found that the github RPM was 20240203 and that worked.  I neglected to change the lua to config.prefer_to_spawn_tabs (20240202 resulted in err)...which does in fact work.

Thanks for the instruction...marking answered,
